aspnetcore/migration/mvc.md

```diff
@@ -3,18 +3,189 @@ title: Migrate from ASP.NET MVC to ASP.NET Core MVC
+This article shows how to get started migrating an ASP.NET MVC project to [ASP.NET Core MVC](../mvc/overview.md). In the process, it highlights related changes from ASP.NET MVC.
+
+Migrating from ASP.NET MVC is a multiple step process and this article covers:
+
+* The initial setup.
+* Basic controllers and views.
+* Static content.
+* Client-side dependencies.
+
+For migrating configuration and identity code, see [Migrate configuration to ASP.NET Core](configuration.md) and [Migrate Authentication and Identity to ASP.NET Core](identity.md).
+
+## Prerequisites
+
+[!INCLUDE [prerequisites](../includes/net-core-prereqs-vs-3.1.md)]
+
+## Create the starter ASP.NET MVC project
+
+Create an example ASP.NET MVC project in Visual Studio to migrate:
+
+1. From the **File** menu, select **New** > **Project**.
+1. Select **ASP.NET Web Application (.NET Framework)** and then select **Next**.
+1. Name the project *WebApp1* so the namespace matches the ASP.NET Core project created in the next step. Select **Create**.
+1. Select **MVC**, and then select **Create**.
+
+## Create the ASP.NET Core project
+
+Create a new solution with a new ASP.NET Core project to migrate to:
+
+1. Launch a second instance of Visual Studio.
+1. From the **File** menu, select **New** > **Project**.
+1. Select **ASP.NET Web Core Web Application** and then select **Next**.
+1. In the **Configure your new project** dialog, Name the project *WebApp1*.
+1. Set the location to a different directory than the previous project to use the same project name. Having the same namespace makes it easier to copy code between the two projects. Select **Create**.
+1. In the **Create a new ASP.NET Core Web Application** dialog, confirm that **.NET Core** and **ASP.NET Core 3.1** are selected. Select the **Web Application (Model-View-Controller)** project template, and select **Create**.
+
+## Configure the ASP.NET Core site to use MVC
+
+When targeting .NET Core, the [Microsoft.AspNetCore.App metapackage](xref:fundamentals/metapackage-app) is referenced by default. This package contains packages commonly used by MVC apps. If targeting .NET Framework, package references must be listed individually in the project file. `Microsoft.AspNetCore.Mvc` is the ASP.NET Core MVC framework.
+
+In ASP.NET Core, the `Startup` class:
+
+* Replaces *Global.asax*.
+* Handles all app startup tasks.
+
+For more information, see <xref:fundamentals/startup>.
+
+* In the ASP.NET Core project, open the *Startup.cs* file.
+
+  [!code-csharp[](mvc/samples/3.x/Startup.cs?highlight=26,43,45)]
```

Comment:
Dedent the code snippet, since the list was eliminated in the suggestion above.

```suggestion
[!code-csharp[](mvc/samples/3.x/Startup.cs?highlight=26,43,45)]
```